### ISSUE TITLE
How to reproduce action segmentation

### ISSUE THREAD

**Issue by laohuijiadezhu** [ttlmh/Bridge-Prompt#13]
It is mentioned in the paper that the features extracted in the "Extracting frame features" (in README.md) section are used as the training input of ASFormer.
I read the source code and README of ASFormer.  I think there are two steps, prediction and evaluation, that need to be run by myself. For prediction I should not be able to directly copy commands. The source code may need to be modified. Where should I modify it? Or is there any other operation to implement?
I'm looking forward to your answer. Thanks!

**Comment by ttlmh** [comment]
Sorry for the late reply! You can refer to #8

**Comment by habakan** [comment]
@ttlmh Hi! Thanks for always your kind advice.
I tried to train BridgePrompt features + ASFormer in 50salads dataset using #8 as a reference.
In my environment, the time-frame dimension was different from I3D datasets (Bridge Prompt features are 7 frames less than I3D).
This is code that works for my environment.    
https://github.com/habakan/ASFormer/blob/bridge_prompt_experiment/batch_gen.py  
```
96:            features = np.load(batch_features[idx]).T[:, :-7]
```
I extracted frames and generated features using README of this repository, did you occur this problem?
Thank you.

**Comment by ttlmh** [comment]
Yes, we do find that frame numbers may not be consistent with that of I3D features. I think the problem lies in video data pre-processing when the frames were extracted. We have not yet found a proper way to extract the same frame numbers as I3D features, but it seems to be harmless to the results.

**Comment by XuanHien304** [comment]
hi @habakan, can I have a code to extract frame-wise feature and label for 50salads dataset

**Comment by habakan** [comment]
@ttlmh 
Thanks for your comments!   
I successfully ran training and testing, but the score was quite low(accuracy is 3 ~ 5%).  
I will contact you if I know anything...!

@XuanHien304 
I didn't do anything special. I extracted Bridge-Prompt features using the command in README.md.
```
python preprocess/get_frames.py --dataset (name_dataset) --vpath (folder_to_your_videos) --fpath ./data/(name_dataset)/frames/
python extract_frame_features.py --config ./configs/(dataset_name)/(dataset_name)_exfm.yaml --dataset (dataset_name)
```
I used label in I3D features provided by MS-TCN feature datasets.
I will share the code after running successfully!

**Comment by XuanHien304** [comment]
@habakan  Did you change the path of pretrained model for extract features for 50 salads? https://github.com/ttlmh/Bridge-Prompt/blob/f97615612f8f557631f3c8d3b6a4293cd4046df8/configs/salads/salads_exfm.yaml#L1

**Comment by habakan** [comment]
@XuanHien304 
Yes. I downloaded the link in readme, and changed the path as [this issue page](https://github.com/ttlmh/Bridge-Prompt/issues/12).

**Comment by laohuijiadezhu** [comment]
update 20221108
The issue has been solved. #16 

***

@habakan 
Hi, have you successfully extracted the features of breakfast and combined them? When I combine the features, I will encounter numpy broadcast errors, such as
```
Traceback (most recent call last):
  File "preprocess/combine_features.py", line 36, in <module>
    result[row.ind:row.ind + 32, :] = tfeat
ValueError: could not broadcast input array from shape (32,768) into shape (21,768)
```
, #16 , etc.
I really need your help and look forward to your answers. Thank you very much!

**Comment by habakan** [comment]
@laohuijiadezhu 
Thanks for sharing your information.  
But I tried only 50salads dataset...! sorry.  
I will share it when I try at breakfast dataset.

**Comment by XuanHien304** [comment]
@habakan  Can you share the code to extract and train model on 50salads dataset

**Comment by clearlove7-star** [comment]
@habakan Hi,do you need to change the split value in datasets.py and salads_exfm.yaml

**Comment by clearlove7-star** [comment]
@habakan Whether the split value needs to be modified when extracting features in datasets.py and salads_exfm.yaml
